README.md

```diff
@@ -211,6 +211,43 @@ originating from tasks on the dispatcher queue. It is however possible that
 `Defect` that happen in tasks bubble up through `poll` as these are not caught
 by the transformation.
 
+#### Checked exceptions
+
+By specifying a `raises` list to an async procedure, you can check which
+exceptions can be thrown by it.
+```nim
+proc p1(): Future[void] {.async, raises: [IOError].} =
```

Comment:
btw, this will break existing code that has raises annotations - we have such code "out there"

Reply:
we might want to consider `asyncraises` instead - this would neatly work around compatibility issues, and would highlight the semantic differences compared to normal "raises"

Reply:
iirc to make nimbus compile, there was around 10 lines changed in the whole codebase, not sure it's worth it to duplicate the entire macro instead of just fixing the 10 lines

Especially since theses 10 lines are greatly error inducing

Reply:
> duplicate the entire macro

the current duplication was borne out of laziness, not necessity - ie one can factor the macro into better quality code to avoid this

Reply:
@arnetheduck to be clear, you're thinking about
`{.async, asyncraises: [].}`
or just
`{.asyncraises: [].}`
?

Reply:
the former - new feature, new keyword (purely additive changes help maintain compatibility generally)

I'd actually also add the capability to print warnings when the asyncraises list is missing - this is something we're unable to do with "imissing" `push raises` for example

Reply:
Done

Reply:
wait, "the former" as in ` {.async, asyncraises: [].}` - the code now does the latter

Reply:
Whoops, forgot my english dictionary at home :sweat_smile: 
Though it's a bit of a mouthful, maybe `{.async, asraises: [].}` or similar would be better?

Reply:
I dunno - I'm fine with the longer name: exceptions are supposed to be exceptional after all. 

The other thing I'd like to arrive at, at some point is a mode where no exceptions are allowed through, similar to `nothrows` in c++, which simply disables all these expensive and redundant code paths altogether (catching and re-raising exceptions is significantly slow, in addition to the "normal" slowness of creating an Exception object and in addition to the unpleasant control flow and cognitive overhead issues exceptions cause). 

The other thing is the `.error` pragma that @zah has been contemplating - to make that smooth (as in, coherent and efficient), we'd probably need to get some help for the language, but once that's been developed to completion, perhaps a single pragma with a single semantic can cover both sync and async. Until then, the longer name is fine and in the unlikely event it _really_ becomes a real bother, we can always add a shorter alias later.

Reply:
All that said, I'm not strongly tied to keeping this a separate keyword: `asyncraises` achieves the same core objective which is to leave existing code working as-is - it does mix two concepts however in a single keyword, which I generally think should be avoided unless there's exceptional (haha) reasons to do it.

Reply:
Should be easy enough to add a fast-path for RaiseTrackingFuture[_, void] that skips the catching-raising dance
And then add a shortcut for `.async, asyncraises: [].`, eg `.neasync.` (no exception async, terrible name, I'm bad at naming)

Reply:
can add that fast path right away, as far as I'm concerned - we can worry about lazy typists later :)

Reply:
in the longest run, what I'd really want is for `asyncraises: []` to be the default when you do `async` - ie that you have to enable and explicitly list all exceptions if you want them - but that's a compatibility nightmare.

Reply:
Fast path is an optimization that can be done later without issues, changing the pragma name is a breaking change

Updated to `.async, asyncraises: [].`, and added `-d:chronosWarnMissingRaises`